tests/unit/test_json.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+import json
+import os.path
+
+JSON_PATHS = [os.path.join('inspirehep', 'modules', 'records', 'jsonschemas'),
+              os.path.join('inspirehep', 'modules', 'workflows', 'mappings'),
+              os.path.join('inspirehep', 'modules', 'records', 'mappings')]
+
+
+def test_pretty_json():
+    for root_dir in JSON_PATHS:
+        for json_dir, _, json_files in os.walk(root_dir):
+            for json_file in json_files:
+                if not json_file.endswith(".json"):
+                    continue
+                json_path = os.path.join(json_dir, json_file)
+                json_content = open(json_path).read()
+                assert json_content == json.dumps(json.loads(json_content), indent=4, sort_keys=True), "{file} is not pretty. Please run ./scripts/prettyjson before committing"
```

Comment:
Exactly, makes no sense waiting 20 mins to find out something you'd see in
a few seconds. If the tests were taking minutes to run then obviously this
would be a no go. But still, overall it's all relative. I would argue that
development would speed up in this way and travis would merely be there as
a sanity check for completely fresh installations.

On Wed, 20 Jul 2016, 09:37 Jacopo Notarstefano, <notifications@github.com>
wrote:

> In tests/unit/test_json.py
> <https://github.com/inspirehep/inspire-next/pull/1351#discussion_r71478177>
> :
>
> > +import os.path
> > +
> > +JSON_PATHS = [os.path.join('inspirehep', 'modules', 'records', 'jsonschemas'),
> > +              os.path.join('inspirehep', 'modules', 'workflows', 'mappings'),
> > +              os.path.join('inspirehep', 'modules', 'records', 'mappings')]
> > +
> > +
> > +def test_pretty_json():
> > +    for root_dir in JSON_PATHS:
> > +        for json_dir, _, json_files in os.walk(root_dir):
> > +            for json_file in json_files:
> > +                if not json_file.endswith(".json"):
> > +                    continue
> > +                json_path = os.path.join(json_dir, json_file)
> > +                json_content = open(json_path).read()
> > +                assert json_content == json.dumps(json.loads(json_content), indent=4, sort_keys=True), "{file} is not pretty. Please run ./scripts/prettyjson before committing"
>
> Hear, hear! I'm actually against this test: since this is still running
> the beautification anyway to check that the JSON looks good, why don't we
> just beautify the JSON in the commit hook?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/1351/files/c2d3fee24addd6b4343cfde67bedaa14aefdaf64#r71478177>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARPHBYY5KDjetmqjFSWrSu9cIoXr6D5ks5qXdBOgaJpZM4JPxDs>
> .
>